pkg/registry/apps/daemonset/strategy.go

```diff
@@ -154,6 +154,7 @@ func daemonSetSurgeFieldsInUse(ds *apps.DaemonSet) bool {
 func (daemonSetStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
 	daemonSet := obj.(*apps.DaemonSet)
 	opts := pod.GetValidationOptionsFromPodTemplate(&daemonSet.Spec.Template, nil)
+	opts.AllowInvalidLabelValueInSelector = false
```

Comment:
why is this hard-coded to false? shouldn't it just use the value computed inside GetValidationOptionsFromPodTemplate?